tasks/insights-client.yml

```
@@ -29,6 +29,13 @@
   when:
     - rhc_insights.tags | d(None) == __rhc_state_absent
 
+- name: Configure autoupdate
+  lineinfile:
+    path: /etc/insights-client/insights-client.conf
+    regexp: "^auto_update"
+    insertafter: "#auto_update"
+    line: "auto_update={{ rhc_insights.autoupdate| d(omit) | bool }}"
```

```suggestion
    line: auto_update={{ rhc_insights.autoupdate | d(omit) | bool }}
```
This seems a little bit tricky.  There are the following cases:
* user does not define `rhc_insights` in their inventory/playbook - then the default value will be used, which means `rhc_insights.autoupdate: true`
* user sets `rhc_insights: {}` or some other empty value - in this case, we do not want to touch insights-client.conf
* user sets `rhc_insights` and sets parameters other than `autoupdate` e.g. `rhc_insights.remediation` - in that case we want to use the default value of `rhc_insights.autoupdate: true`
* user explicitly sets `rhc_insights.autoupdate: $value`

note that if the user does something like `rhc_insights.autoupdate: none` or some value that is not `true` or `false`, we should use the `| bool` filter to attempt to convert the value to a boolean, and let the role fail if the value cannot be converted to `true` or `false`

---

Yes it seems: I opted to change the `omit` to `true`. That way the playbook assures that is `true` by default, whether is not present or explicit expressed. (Which is also the default value of `insights-client`)